### PR TITLE
store all solutions

### PR DESCRIPTION
**Pull request by Wikunia**
Close: #26 
Storing all solutions and have them sorted by objective value at the very end of the optimize step.

### CODE REVIEW

**Comment by Wikunia** [comment]
Would you consider this already be helpful enough as a first version @matbesancon ?

File: src/Bonobo.jl [hunk 1]
```diff
@@ -351,17 +361,12 @@ end
...
 This function needs to be implemented by you if you have a different type of Solution object than [`DefaultSolution`](@ref).
 """
 function add_new_solution!(tree::BnBTree{N,R,V,S}, node::AbstractNode) where {N,R,V,S<:DefaultSolution{N,V}}
     sol = DefaultSolution(node.ub, get_relaxed_values(tree, node), node)
-    if isempty(tree.solutions)
-        push!(tree.solutions, sol)
-    else
-        tree.solutions[1] = sol
-    end
+    push!(tree.solutions, sol)
```
Comment (matbesancon) [inline src/Bonobo.jl:369]: it doesn't get sorted here?

Comment (Wikunia) [reply src/Bonobo.jl:369]: No it does get sorted at the end of `optimize!` before it's in the order in which they are found. I could also sort them here but wasn't sure it's faster and it seemed a bit more complicated. In the search process only the actual incumbent is relevant, right?

Comment (matbesancon) [reply src/Bonobo.jl:369]: yes but the current feasible vector corresponding to the incumbent might be relevant

Comment (Wikunia) [reply src/Bonobo.jl:369]: I could have an `incumbent_solution` field or have them sorted here.

Comment (Wikunia) [reply src/Bonobo.jl:372]: Added that for now.

**Pull request merged**